chanzuckerberg/idseq-web
# IDSEQ-1398: Ensure required metadata fields (and host defaults) are associated with all host genomes

# Description

In IDSEQ-1398, a user was unable to upload because their host (bat) did not have required field `collection_location_v2`. This adds model validation and a callback to ensure this can't happen again.

In addition, I will run the following script in prod to update existing hosts.

```
  MetadataField.where(is_required: 1).each do |mf|  
    mf.update_host_genomes!
    mf.save!
  end

  MetadataField.where(default_for_new_host_genome: 1).each do |mf|  
    mf.update_host_genomes!
    mf.save!
  end
```

# Notes

* This also raises the question of whether we should update existing projects on adding a required field

# Tests

* try creating a required field with no default_for_new_host_genome
* try updating to a required field with no default_for_new_host_genome
* try creating a new field that is required, see hosts update
